src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DiagnosticReportStatusTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class DiagnosticReportStatusTest {
+
+    private static final String VALID_EDIFACT = "STS++UN";
+    private static final String VALID_EDIFACT_VALUE = "UN";
+
+
+    @Test
+    void testToEdifactWithValidDiagnosticReportStatus() {
+        final String expected = "STS++UN'";
+
+        final DiagnosticReportStatus diagnosticReportStatus = DiagnosticReportStatus.builder()
+            .event(ReportStatusCode.UNSPECIFIED)
+            .build();
+
+        final String actualValue = diagnosticReportStatus.toEdifact();
+
+        assertEquals(expected, actualValue);
+    }
+
+    @Test
+    void testToEdifactWithDetailsInDiagnosticReportStatus() {
+        final String expected = "STS+Details+UN'";
+
+        final DiagnosticReportStatus diagnosticReportStatus = DiagnosticReportStatus.builder()
+            .detail("Details")
+            .event(ReportStatusCode.UNSPECIFIED)
+            .build();
+
+        final String actualValue = diagnosticReportStatus.toEdifact();
+
+        assertEquals(expected, actualValue);
+    }
+
+    @Test
+    void testBuildWithNullEventThrowsException() {
+        assertThrows(NullPointerException.class, () -> DiagnosticReportStatus.builder().build());
+    }
+
+    @Test
+    void testFromStringWithValidEdifactStringReturnsDiagnosticReportStatus() {
+        final DiagnosticReportStatus diagnosticReportStatus = DiagnosticReportStatus.fromString(VALID_EDIFACT);
+
+        assertEquals("STS", diagnosticReportStatus.getKey());
+        assertEquals(VALID_EDIFACT_VALUE, diagnosticReportStatus.getValue());
+        assertEquals("STS++UN'", diagnosticReportStatus.toEdifact());
```

Comment:
When making multiple assertions, I recommend using `assertAll(...)` so that if more than one of them ever fail, we are told about all failures instead of just the first one.
```suggestion
        assertAll(
            () -> assertEquals("STS", diagnosticReportStatus.getKey()),
            () -> assertEquals(VALID_EDIFACT_VALUE, diagnosticReportStatus.getValue()),
            () -> assertEquals("STS++UN'", diagnosticReportStatus.toEdifact())
        );
```